protoarray/src/main/java/tech/pegasys/teku/protoarray/ProtoNode.java

```diff
@@ -122,4 +123,50 @@ public void setBestChildIndex(Optional<Integer> bestChildIndex) {
   public void setBestDescendantIndex(Optional<Integer> bestDescendantIndex) {
     this.bestDescendantIndex = bestDescendantIndex;
   }
+
+  public ProtoNode cloneForSaving() {
```

Comment:
(nit) Why not just:
```suggestion
  public ProtoNode clone() {
```

Reply:
Because I don't clone variables that are not necessary for saving such as parentIndex, weight etc.